docs/components/integration-framework/connectors/out-of-the-box-connectors/googledrive.md

```diff
@@ -0,0 +1,245 @@
+
+**WARNING!** Following code snippets are for demonstration purposes only and must be not used for real production systems due to security concerns.
+
+Assuming you have created a service account and downloaded a JSON file with keys. Run the following code snippet that prints the JWT token in terminal.
+
+```python
+import google.auth
+import google.auth.transport.requests
+from google.oauth2 import service_account
+# Scopes required to execute 'create' endpoind with Google Drive API
+SCOPES = ['https://www.googleapis.com/auth/drive', 'https://www.googleapis.com/auth/drive.file', 'https://www.googleapis.com/auth/drive.appdata']
+# File with keys
+SERVICE_ACCOUNT_FILE = 'google-service-account-creds.json'
+credentials = service_account.Credentials.from_service_account_file(SERVICE_ACCOUNT_FILE, scopes=SCOPES)
+auth_req = google.auth.transport.requests.Request()
+credentials.refresh(auth_req)
+# Print token
+print(credentials.token)
+```
+
+#### Example 2: obtaining bearer and refresh tokens with OAuth client
+
+![Refresh Auth](../img/connectors-googledrive-jwt-refresh.png)
+
+See [related article](https://developers.google.com/identity/protocols/oauth2/web-server).
+
+**WARNING!** Following code snippets are for demonstration purposes only and must be not used for real production systems due to security concerns.
+
+Assuming you have created an OAuth client, you can download key files from the Google [Console](https://console.cloud.google.com/apis/credentials). Run the following code snippet that prints the refresh token in terminal.
+
+```python
+from google_auth_oauthlib.flow import InstalledAppFlow
+import pprint
+
+SCOPES = ['https://www.googleapis.com/auth/drive', 'https://www.googleapis.com/auth/documents']
+OAUTH_KEYS = './oauth-keys.json' # path to your file with OAuth credentials
+
+def main():
+    flow = InstalledAppFlow.from_client_secrets_file(OAUTH_KEYS, SCOPES)
+    creds = flow.run_local_server(port=54948)
+    pprint.pprint(vars(creds))
+
+if __name__ == "__main__":
+    main()
+```
+
+### Where do I get a parent folder ID?
+
+It is in the URL.
+
+![Google Drive Connector get parent folder ID](../img/connectors-googledrive-get-parent-folder-id.png)
+
+### How do I set additional properties or metadata?
+
+You can set any property from the Google Drive [Create API](https://developers.google.com/drive/api/v3/reference/files/create).
+For example:
+
+```
+= {
+  "description":"myDescription"
+}
+```
+
+The unknown or mistyped properties will be ignored.
+
+### What are the limitations of the additional properties or metadata?
+
+Some properties are applicable only for the token owners, like `folderColorRgb` and `starred`.
+
+### Where do I get Template ID?
+
+It is in the URL.
+
+![Google Drive Connector get template ID](../img/connectors-googledrive-get-template-id.png)
+
+### Can you show me an example of a valid template?
+
+The valid template looks like the one on a screenshot.
```

Comment:
Pls update text to: "A valid template looks like the following screenshot."

Reply:
Addressed in a new revision. Will publish shortly.